### PR TITLE
Feature/admin destinations crud

### PR DESCRIPTION
**Pull request by AlexPop28**
This PR depends on #6.
Closes https://trello.com/c/ZiYwWofu

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -14,7 +22,7 @@ const router = createBrowserRouter([
   },
   {
     path: "/login",
-    element: <LoginPage />,
+    element: <Login />,
```
Comment (Laurcons) [inline src/App.tsx]: Same comment here

File: src/App.tsx [hunk 2]
```diff
@@ -24,13 +32,43 @@ const router = createBrowserRouter([
...
+  {
+    path: "/admin/destinations/add",
+    element: <AddDestinationPage />,
+  },
+  {
+    path: "/admin/destinations/:destinationId",
+    element: <EditDestinationPage />,
+  },
+  {
+    path: "/page-not-found",
```
Comment (Laurcons) [inline src/App.tsx:60]: Does this work as a 404 page? Are components redirecting to this by any chance?

File: src/components/Layout.tsx [hunk 3]
```diff
@@ -25,36 +25,54 @@ const links: LinkData[] = [
...
+  },
 ];
 
 export default function Layout({ children }: PropsWithChildren) {
-  const role = Cookies.get("role");
+  const { isLoading, user, logout } = useUser();
+
   return (
     <>
       <Title>Mama are mere</Title>
```
Comment (Laurcons) [inline src/components/Layout.tsx]: Probably want to make this a prop

File: src/pages/admin/users/AddUser.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  const handleAdd = async () => {
+    const response = await axios.post("/admin/users", {
+      email,
+      password,
+      role,
+    });
+
+    if (response.status != 201) {
+      throw Error("There was a problem and the user was not added.");
```
Comment (Laurcons) [inline src/pages/admin/users/AddUser.tsx]:
```suggestion
      throw new Error("There was a problem and the user was not added.");
```

File: src/pages/admin/users/EditUser.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import { useQuery } from "react-query";
+import { useNavigate, useParams } from "react-router";
+import Button from "../../../components/Button";
+import Layout from "../../../components/Layout";
+import UserForm from "../../../components/UserForm";
+import useAxios from "../../../lib/hooks/useAxios";
+import { useAdmin } from "../../../lib/hooks/useRole";
+
+const EditUser = () => {
+  useAdmin();
```
Comment (Laurcons) [inline src/pages/admin/users/EditUser.tsx]: Please rename to be more self-explanatory

```suggestion
  useRequireAdmin(); // or useRequireAdminRole()
```

**Review by Laurcons — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 2 above)
Comment (AlexPop28) [reply src/App.tsx:60]: Yes, more specifically when the user does not have the necessary role to view the page.

**Review by Laurcons — APPROVED** [review]

**Pull request merged**